pkg/controllers/v1/mpi_job_controller.go

```diff
@@ -1443,3 +1457,17 @@ func isCleanUpPods(cleanPodPolicy *common.CleanPodPolicy) bool {
 	}
 	return false
 }
+
+func isGPULauncher(mpiJob *kubeflow.MPIJob) bool {
+	replica, ok := mpiJob.Spec.MPIReplicaSpecs[kubeflow.MPIReplicaTypeLauncher]
+	if !ok {
+		klog.Warningf("MPIJob %s: Launcher not found in mpijob spec", mpiJob.Name)
+		return false
+	}
+	for _, container := range replica.Template.Spec.Containers {
+		if _, ok := container.Resources.Requests[gpuResourceName]; ok {
```

Comment:
Currently this only supports NVDIA GPUs. Can we relax that to something like gpuResourceNameRegex?

Reply:
Sure，could we use `gpuResourceNameSuffix=.com/gpu` according to the doc https://kubernetes.io/docs/tasks/manage-gpus/scheduling-gpus/#using-device-plugins.

And change `Resources.Requests` to `Resources.Limits` because `GPUs are only supposed to be specified in the limits section`。

Reply:
Sounds good

Reply:
done